jenkins-scripts/dsl/gz-collections.yaml

```diff
@@ -445,6 +445,10 @@ collections:
         major_version: 3
         repo:
           current_branch: main
+      - name: gz-cmake
+        major_version: 4
```

Comment:
should this be 5?

Reply:
We don't have a branch with that yet. If we want to this to be 5, I think `main` will need to be bumped. I was hoping not to go through version bumps because it's a lot of effort.

Reply:
this is the `__upcoming__` section, which currently only has `gz-tools3`, so I suggested bumping the major version based on pattern matching, not based on how I think `__upcoming__` should be used. This is fine as is